### DISCUSSION TITLE
Toolbar user feedback

### DISCUSSION THREAD

**Discussion by SMERKIN5000** [voxeltycoon/issues#789] [UI&UX]
I've really tried to embrace the in-game toolbar keys (CTRL + 1, CTRL + 2 etc) however my brain just doesn't work this way and I've abandoned using them. Mainly there are several things I do frequently, like building a road, a railway, specific signals and I find that navigating in and out of the Toolbar levels not as efficient as they could be.

Here's an example. CTRL + 1 is the first hotbar item. If I want to build a rail then I have to do CTRL + 3 (Rails) then CTRL + 1 (Lay Rails). Then, I decide I want to lower the terrain, so that is CTRL + 5 HOWEVER I need to escape back out of the rails menu to access the terrain menu. Pressing escape only closes the (Lay Rails) build menu and does not deselect the rails menu and pressing escape too many times starts to close windows I have open. I went to the extent of stetting up macros to help with this but I've ended up not using them as it doesn't really help me.

If I use another game as an example, Satisfactory utilities a great customizable toolbar experience for the player. There a 0-9 hotkey slots, so when I'm in the build menu and I hover on a build-able item, if I press 6 it is assigned to toolbar slot 6. Then in game pressing 6 starts building this item.

In addition to this, Satisfactory has 10 separate hotbars you can scroll through, ALT + Scrollwheel allows you to rotate through them.

**Comment by tycode** [comment]
I've reassigned the toolbar keys to 1, 2, 3 etc. (without ctrl) but they're still virtually unusable due to the inability to easily get back to the main toolbar.

I'd love if it was done similarly to Mindustry: the key always presses the button on the _main_ toolbar, unless it's pressed in quick succession after another toolbar key, in which case it operates on the sub toolbar instead. For example, 3, 1 in quick succession would always lay rails, _regardless_ of which toolbar (if any) was open previously. 1, 1 in quick succession would build mines. Pressing 3 on its own would deselect any selected tool and open the rail toolbar with no tool selected.
